p2p/src/peer_manager/mod.rs

```diff
@@ -790,7 +795,14 @@ where
         }
     }
 
-    fn handle_add_list_request(&mut self, peer: PeerId) {
+    fn handle_add_list_request(&mut self, peer_id: PeerId) {
+        let peer = self.peers.get_mut(&peer_id).expect("peer must be known");
+        if *peer.ignore_addr_list_request {
+            log::warn!("Ignore unexpected address list request from peer {peer_id}");
+            return;
+        }
+        peer.ignore_addr_list_request.set();
```

Comment:
Two thing I can thing to improve this:

1. This can be isolated into its own function
2. SetFlag can have a test_and_set method that both returns true and sets the value similar to the way C++ does it for atomics: https://cplusplus.com/reference/atomic/atomic_flag/test_and_set/

If No. 2 is implemented, No. 1 isn't necessary. The problem with its current form is that it's kind of weird how it's done, where we check then manually set a value. The whole degrees-of-freedom discussion can be brought up again.